niworkflows/viz/plots.py

```diff
@@ -1036,28 +1035,67 @@ def get_surface_meshes(density, surface_type):
     # as potential nonsteady states
     data = img.dataobj[5:20].mean(axis=0)
 
-    cortex_data = _concat_brain_struct_data((left_cortex, right_cortex), data)
-    if density == "32k" and len(cortex_data) != 59412:
+    counts = (left_cortex.index_count, right_cortex.index_count)
+    if density == "32k" and counts != (29696, 29716):
         raise ValueError("Cortex data is not in fsLR space")
+
     # medial wall needs to be added back in
-    cortex_data = add_fslr_medial_wall(cortex_data)
+    lh_data = np.full(left_cortex.surface_number_of_vertices, np.nan)
+    rh_data = np.full(right_cortex.surface_number_of_vertices, np.nan)
+    lh_data[left_cortex.vertex_indices] = _concat_brain_struct_data([left_cortex], data)
+    rh_data[right_cortex.vertex_indices] = _concat_brain_struct_data([right_cortex], data)
+
     if clip_range:
-        cortex_data = np.clip(cortex_data, clip_range[0], clip_range[1], out=cortex_data)
+        lh_data = np.clip(lh_data, clip_range[0], clip_range[1], out=lh_data)
+        rh_data = np.clip(rh_data, clip_range[0], clip_range[1], out=rh_data)
+        mn, mx = clip_range
+    else:
+        mn, mx = None, None
 
-    lh_data, rh_data = np.array_split(cortex_data, 2)
+    if mn is None:
+        mn = np.min(data)
+    if mx is None:
+        mx = np.max(data)
```

Comment:
I'm realizing that this is including the min/max of subcortical grayordinates, which is why things got a little less bright. I think I prefer using the full data range like this, but let me know if you'd rather it be based purely on `lh_data` and `rh_data`.